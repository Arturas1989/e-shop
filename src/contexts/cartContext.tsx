import { ReactNode, createContext, useContext } from "react";
import { Cart } from "../services/Cart";
import { useCart } from "../hooks";

type CartContextProviderProps = {
  children: ReactNode;
};

type CartContextValue = {
  cart: Cart | null;
  setCart: React.Dispatch<React.SetStateAction<Cart | null>>;
  isLoading: boolean;
} | null;

const CartContext = createContext<CartContextValue>(null);

export const CartContextProvider = ({ children }: CartContextProviderProps) => {
  const {cart, setCart, isLoading} = useCart();
  return (
    <CartContext.Provider value={{cart, setCart, isLoading}}>
      {children}
    </CartContext.Provider>
  )
};

export const useCartContext = () => useContext(CartContext);
