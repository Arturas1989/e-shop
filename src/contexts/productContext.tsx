import { ReactNode, createContext, useContext } from "react";
import { DataInfo, Product } from "../types";
import { useData } from "../hooks";

type ProductContextProviderProps = {
  children: ReactNode;
};

type ProductContextValue = {
  products: Product[] | null;
  setProducts: React.Dispatch<React.SetStateAction<Product[] | null>>;
  isProductsLoading: boolean;
  featuredProducts: Product[] | null;
  setFeatured: React.Dispatch<React.SetStateAction<Product[] | null>>;
  isFeaturedProductsLoading: boolean;
} | null;

const ProductContext = createContext<ProductContextValue>(null);

export const ProductContextProvider = ({ children }: ProductContextProviderProps) => {
  const productsInfo: DataInfo = {
    collectionName: 'products',
    onlyFeatured: false,
  }
  const [products, isProductsLoading, setProducts] = useData<Product>(productsInfo);
  const featuredInfo: DataInfo = {
    collectionName: 'products',
    onlyFeatured: true
  }
  const [featuredProducts, isFeaturedProductsLoading, setFeatured] = useData<Product>(featuredInfo);

  const contextVal = {
    products, isProductsLoading, setProducts, featuredProducts, setFeatured, isFeaturedProductsLoading
  }
  return (
    <ProductContext.Provider value={contextVal}>
      {children}
    </ProductContext.Provider>
  )
};

export const useProductsContext = () => useContext(ProductContext);
