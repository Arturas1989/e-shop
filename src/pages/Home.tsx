export const Home = () => {
  return (
    <main data-testid='home'>Home</main>
  )
};
