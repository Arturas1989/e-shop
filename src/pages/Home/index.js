export { Hero } from './components/Hero/Hero';
export { FeaturedProducts } from './components/FeaturedProducts/FeaturedProducts';
export { Testimonials } from './components/Testimonials/Testimonials';