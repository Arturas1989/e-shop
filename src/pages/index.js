export { Home } from './Home/Home';
export { ProductsPage } from './Products/ProductsPage';
export { PageNotFound } from './PageNotFound/PageNotFound';

export { Register } from './Auth/Register/Register';
export { Login } from './Auth/Login/Login';
