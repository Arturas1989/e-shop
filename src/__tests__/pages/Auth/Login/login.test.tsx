import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Login } from '../../../../pages';
import { CartContextProvider } from '../../../../contexts/cartContext';

describe('Login page test', () => {
  test('render test', () => {
    const tree = renderer
      .create(
        <CartContextProvider>
          <BrowserRouter>
            <Login />
          </BrowserRouter>
        </CartContextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
