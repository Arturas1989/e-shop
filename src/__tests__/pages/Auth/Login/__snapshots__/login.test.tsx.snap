// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page test render test 1`] = `
<main
  className="main login"
>
  <section
    className="main-section"
  >
    <h2
      className="heading"
    >
      Login
    </h2>
    <form
      className="main-form"
      onSubmit={[Function]}
    >
      <div
        className="form-field"
      >
        <label
          htmlFor="email"
        >
          Your email
        </label>
        <input
          autoComplete="off"
          className="input-bar"
          id="email"
          placeholder="Enter your email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="form-field"
      >
        <label
          htmlFor="password"
        >
          Your password
        </label>
        <input
          autoComplete="off"
          className="input-bar"
          id="password"
          placeholder="Enter your password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="secondary-button"
        onClick={[Function]}
      >
        Login
      </button>
    </form>
  </section>
</main>
`;
